pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -0,0 +1,437 @@
+	}
+
+	srcContext, err := findNewest(isTags)
+	if err != nil {
+		return fmt.Errorf("failed to find the new isTag from %v: %w", isTags, err)
+	}
+	sourceImageStreamTag := isTags[srcContext]
+
+	imageStreamNameAndTag := strings.Split(decoded.Name, ":")
+	if n := len(imageStreamNameAndTag); n != 2 {
+		return fmt.Errorf("when splitting imagestreamtagname %s by : expected two results, got %d", decoded.Name, n)
+	}
+	imageStreamName, imageTag := imageStreamNameAndTag[0], imageStreamNameAndTag[1]
+	isName := types.NamespacedName{Namespace: decoded.Namespace, Name: imageStreamName}
+	sourceImageStream := &imagev1.ImageStream{}
+	if err := r.registryClients[srcContext].Get(r.ctx, isName, sourceImageStream); err != nil {
+		return fmt.Errorf("failed to get imageStream %s from registry cluster: %w", isName.String(), err)
+	}
+
+	*log = *log.WithField("docker_image_reference", sourceImageStreamTag.Image.DockerImageReference)
+	if isImportForbidden(sourceImageStreamTag.Image.DockerImageReference, r.forbiddenRegistries) {
+		log.Debugf("Import from any cluster in %s is forbidden, ignoring", r.forbiddenRegistries)
+		return nil
+	}
+
+	for ctx, isTag := range isTags {
+		if ctx == srcContext {
+			continue
+		}
+		client := r.registryClients[ctx]
+		if err := client.Get(r.ctx, types.NamespacedName{Name: decoded.Namespace}, &corev1.Namespace{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed to check if namespace %s exists: %w", decoded.Namespace, err)
+			}
+			if err := client.Create(r.ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: decoded.Namespace}}); err != nil && !apierrors.IsAlreadyExists(err) {
+				return fmt.Errorf("failed to create namespace %s: %w", decoded.Namespace, err)
+			}
+		}
+
+		imageStreamImport := &imagev1.ImageStreamImport{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: decoded.Namespace,
+				Name:      imageStreamName,
+			},
+			Spec: imagev1.ImageStreamImportSpec{
+				Import: true,
+				Images: []imagev1.ImageImportSpec{{
+					From: corev1.ObjectReference{
+						Kind: "DockerImage",
+						Name: publicURLForImage(sourceImageStreamTag.Image.DockerImageReference),
+					},
+					To: &corev1.LocalObjectReference{Name: imageTag},
+					ReferencePolicy: imagev1.TagReferencePolicy{
+						Type: imagev1.LocalTagReferencePolicy,
+					},
+				}},
+			},
+		}
+
+		r.log.Debugf("creating import %v on cluster %s", imageStreamImport, ctx)
+		if r.dryRun {
+			continue
+		}
+
+		if err := r.ensureCIOperatorRoleBinding(decoded.Namespace, client, log); err != nil {
+			return fmt.Errorf("failed to ensure rolebinding: %w", err)
+		}
+		if err := r.ensureCIOperatorRole(decoded.Namespace, client, log); err != nil {
+			return fmt.Errorf("failed to ensure role: %w", err)
+		}
+		if err := r.ensureImageStream(sourceImageStream, client, log); err != nil {
+			return fmt.Errorf("failed to ensure imagestream: %w", err)
+		}
+
+		if isTag != nil && isTag.Image.Name == sourceImageStreamTag.Image.Name {
```

Comment:
The ImageStreamTagImport should not be defined before this check and we should not log that we might create an import when we probably won't. The client itself has dryRun mode enabled btw, it doesn't work on api.ci because its too old (it will just error) but it works fine on 4.X clusters. So there is no need to have a `dryRun` config option